sunpy/map/mapbase.py

```diff
@@ -672,7 +672,8 @@ def top_right_coord(self):
         """
         The physical coordinate at the center of the the top left ([-1, -1]) pixel.
         """
-        return self.pixel_to_world(*self.dimensions)
+        return self.pixel_to_world(self.dimensions[0] - 1 * u.pix,
+                                   self.dimensions[1] - 1 * u.pix)
```

Comment:
I *think* an easier fix might be
```suggestion
        return self.pixel_to_world(*self.dimensions, origin=1)
```